src/stan/analyze/mcmc/compute_effective_sample_size.hpp

```diff
@@ -35,6 +37,30 @@ namespace analyze {
       num_draws = std::min(num_draws, sizes[chain]);
     }
 
+    if ( num_draws < 3 ) {
+      return std::numeric_limits<double>::quiet_NaN();
+    }
+
+    Eigen::VectorXd draw_val = Eigen::VectorXd::Random(num_chains);
```

Comment:
`draw_val` needs to be documented.  The random initialization is also dangerous as it can yield non-reproducible results -- I recommend going with (1, 2, 3, ..., num_chains) to give distinct but determinism values.

Reply:
(1, 2, 3, ... num_chains), check.  Would you like an inline comment or for `draw_val` to be mentioned in the function documentation?

Reply:
Typically inline if it doesn't affect the outcome of the function evaluation (and hence is just an implementation detail) and in the function doc if it does affect output values so that the client is aware of any expected edge cases.